src/engine/polyline_compressor.cpp

```diff
@@ -61,24 +61,25 @@ std::vector<util::Coordinate> decodePolyline(const std::string &geometry_string)
 
     while (index < len)
     {
-        int b, shift = 0, result = 0;
-        do
+        int b = 0x20, shift = 0, result = 0;
+        while (b >= 0x20 && index < len)
         {
             b = geometry_string.at(index++) - 63;
             result |= (b & 0x1f) << shift;
             shift += 5;
-        } while (b >= 0x20 && index < len);
+        }
         int dlat = ((result & 1) != 0 ? ~(result >> 1) : (result >> 1));
         lat += dlat;
 
+        b = 0x20;
         shift = 0;
         result = 0;
-        do
+        while (b >= 0x20 && index < len)
         {
             b = geometry_string.at(index++) - 63;
```

Comment:
while we are touching this, do you think we could also refactor the loop to use lamdas? Both while loops do the same, right?